src/main/java/frc/robot/commands/TrackTarget.kt

```diff
@@ -1,72 +1,210 @@
+        return {t: Double  -> ( (vx(t) + vy(t)) / (2*sqrt( x(t).pow(2) + y(t).pow(2) ) ) ) }
+    }
+
+    init {
+        addRequirements(Turret)
+    }
 
-    private var samples: Array<Measurement?> = Array(TrackClosest.SAMPLE_NUM){ null }
 
-    fun addMeasurement(measurement: Measurement){
+    //iteratively find roots of function by offsetting a seed based on x intercept of tangent line of last guess
+    //TODO account for chaotic fractal thingie
+    private fun newtonRaphson(seed: Double, func: (Double) -> Double, funcPrime: (Double) -> Double ): Double{
 
-        samples = samples.mapIndexed { index, _ -> if(index != 7) { samples[index+1]} else {measurement}}.toTypedArray()
+        var nextSeed = seed
+        var h = func(nextSeed)/funcPrime(nextSeed)
+
+        while(abs(h) >= AimingConstants.EPSILON){
+
+            h = func(nextSeed)/funcPrime(nextSeed)
+            nextSeed -= h
+
+        }
+        return nextSeed
     }
 
-    fun reset(){
-        samples = Array(TrackClosest.SAMPLE_NUM){ null }
+    private fun constructTrajectory(){
+        //samples
+
+        //offset timestamps so 0 is the current time
+        val smaples = setSampleEpoch(TargetVision.smaples, TargetVision.curTime)
+
+
+        //Generalized Linear Model regresses points to find quadratic functions for translation x and y in terms of t
+        val samplerX: Sampler.DataSampleBuilder = Sampler()
+            .forColumn("t").generate{_, index -> smaples[index].timestamp}
+            .forColumn("t^2").generate{_, index -> smaples[index].timestamp.pow(2)}
+            .forColumn("x/y").generate {_, index -> smaples[index].pose.x}
+            .end()
+
+        val samplerY: Sampler.DataSampleBuilder = Sampler()
+            .forColumn("t").generate{_, index -> smaples[index].timestamp}
+            .forColumn("t^2").generate{_, index -> smaples[index].timestamp.pow(2)}
+            .forColumn("x/y").generate {_, index -> smaples[index].pose.y}
+
+            .end()
+
+        var frameX: DataFrame = schema.build()
+        var frameY: DataFrame = schema.build()
+
+        frameX = samplerX.sample(frameX, AimingConstants.SAMPLE_NUM)
+        frameY = samplerY.sample(frameY, AimingConstants.SAMPLE_NUM)
+
+
+        glm.solverType = GlmSolverType.GlmIrls
+
+        glm.fit(frameX)
+        CoefficientsList.X.coefficients.values = glm.coefficients.values
+
+        glm.fit(frameY)
+        CoefficientsList.Y.coefficients.values = glm.coefficients.values
+
+        val x = quadraticBuilder(CoefficientsList.X.coefficients)
+        val y = quadraticBuilder(CoefficientsList.Y.coefficients)
+        val vx = quadraticDerivativeBuilder(CoefficientsList.X.coefficients)
+        val vy = quadraticDerivativeBuilder(CoefficientsList.Y.coefficients)
+        val tofDerivative = quadraticDerivativeBuilder(CoefficientsList.Tof.coefficients)
+
+        val lhsfunction: (t: Double) -> Double = { t -> pointsToDistance.compose(timeOfFlight)(x(t), y(t))}
```

Comment:
Rename to something unhelpful like `solveFunction` and add a helpful comment linking to docs.